### PR TITLE
[feat] edit adopter name, state, and veteran status

### PR DESCRIPTION
**Pull request by saitunguturu**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- users can edit first name, last name, state, and veteran status
- displays current user info in the text boxes, which can be modified and saved by the user
- accessible after clicking Edit Profile on the profile page

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 
Current Profile page with Edit Profile button to a new page:
image

Edit Profile Page:
image

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
-can be reviewed all at once, simply ensure that the user's info is being correctly retrieved and that the table is being updated when the user clicks the save button after changing info on the edit profile page

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ethan-tam33

### CODE REVIEW

**Comment by ethan-tam33** [comment]
@saitunguturu can you rebase?

File: app/(main)/profile/edit/page.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { getSupabaseServerClient } from '@/lib/supabase/server';
+
+export default async function EditProfilePage() {
+  const supabase = await getSupabaseServerClient();
+  const {
+    data: { user },
+  } = await supabase.auth.getUser();
+
+  if (!user) {
+    return <div>Please log in to edit your profile.</div>;
```
Comment (ethan-tam33) [inline app/(main)/profile/edit/page.tsx]: can you route to the login page instead

File: utils/schema.d.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 export type UUID = string;
 
 export interface Profile {
-  user_id: UUID;
+  user_id: string;
```
Comment (ethan-tam33) [inline utils/schema.d.ts]: is there a reason why this is changed to be a string instead of UUID?

File: components/EditProfilePage.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    user_id: profile?.user_id ?? ('' as string),
+    first_name: profile?.first_name ?? '',
+    last_name: profile?.last_name ?? '',
+    state: profile?.state ?? '',
+    veteran_status: profile?.veteran_status ?? false,
+  });
+
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [success, setSuccess] = useState(false);
```
Comment (jinkang-0) [inline components/EditProfilePage.tsx]: Instead of states, I would suggest to use `useForm` from the [`react-hook-form`](https://react-hook-form.com/) library instead. You can find examples from the components in `components/application/main/...`

File: components/EditProfilePage.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        placeholder="First Name"
+        className="rounded border p-2"
+      />
+      <input
+        name="last_name"
+        value={formData.last_name}
+        onChange={handleChange}
+        placeholder="Last Name"
+        className="rounded border p-2"
+      />
```
Comment (jinkang-0) [inline components/EditProfilePage.tsx]: We should have a custom `<Textbox />` component that comes styled with variants already. It's preferred to use that to keep all text inputs synced in the future.

File: app/(main)/profile/edit/page.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  const supabase = await getSupabaseServerClient();
+  const {
+    data: { user },
+  } = await supabase.auth.getUser();
+
+  if (!user) {
+    redirect('/login');
+  }
+
+  const profile = await fetchProfileById(user.id);
```
[the remark this answers is not in the document]
Comment (jinkang-0) [reply app/(main)/profile/edit/page.tsx]: This allows the page component to remain server-side.

File: app/(main)/profile/page.tsx [hunk 6]
```diff
@@ -18,6 +19,8 @@ export default function ProfilePage() {
     <div className="flex h-full w-full flex-col items-center justify-center">
       <CustomLink href="/">← Go back</CustomLink>
       <p>Profile page</p>
+      <ProfileTest />
+      <button onClick={() => router.push('/profile/edit')}>Edit Profile</button>
```
Comment (jinkang-0) [inline app/(main)/profile/page.tsx]: Since all this button does is navigate to another route, this button can be turned into a `<ButtonLink />` component.

File: tests/ProfileTest.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
+'use client';
+
+import React from 'react';
+import { useProfile } from '@/contexts/ProfileProvider'; // adjust path if needed
```
Comment (jinkang-0) [inline tests/ProfileTest.tsx]: Can you explain what this comment means? If we don't need it, we can probably remove it.

File: components/EditProfilePage.tsx [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+      onSubmit={handleSubmit(onSubmit)}
+      className="flex w-80 flex-col gap-3"
+    >
+      <input
+        name="first_name"
+        defaultValue={profile?.first_name ?? ''}
+        placeholder="First Name"
+        className="rounded border p-2"
+      />
+      <input
```
[the remark this answers is not in the document]
Comment (jinkang-0) [reply components/EditProfilePage.tsx]: I would actually recommend using the `register` function, which you can decompose from the `useForm` hook. It can be passed into the `input` element as a spread prop (i.e. `{...register}`). There are some examples you can find in the components folder, specifically for onboarding or application forms.

The Controller component is a necessary approach for the `react-select` component though, since it's not as clear-cut as a regular HTML input element.

Another note is that I would also recommend using the `<Textbox />` component that we have in the components folder already. This helps standardize textbox styling across the app.

File: components/EditProfilePage.tsx [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+        {loading ? 'Saving...' : 'Save'}
+      </button>
+
+      {success && <p className="text-green-600">Profile updated!</p>}
+      {error && <p className="text-red-600">{error}</p>}
+    </form>
+  );
+}
+
+// 'use client';
```
Comment (ethan-tam33) [inline components/EditProfilePage.tsx]: can we remove this unused code?

**Review by ethan-tam33 — CHANGES_REQUESTED** [review]
thanks Sai, just address the rest of the comments and this should be gtg!

**Review by ethan-tam33 — APPROVED** [review]
